Errors when using Cake Step from the Release Management

<!--
BEFORE YOU SUBMIT AN ISSUE:

DO NOT CREATE AN ISSUE FOR A QUESTION - questions are better served in chat, and can be later raised as an issue if required.
-  chat - https://gitter.im/cake-build/cake

DELETE EVERYTHING IN THIS COMMENT BLOCK

TEMPLATE FOR BUG REPORTS:
-->

### What You Are Seeing?

The step execution is failing during the Cake.ps1 tools resolution

### What is Expected?

To download all the requirement and run, just like when used for the Builds.

### What version of Cake are you using?

v0.3.0

### Are you running on a 32 or 64 bit system?

64 bits

### What environment are you running on?  Hosted? Custom?

Hosted

<!--
If possible, provide a link to the failing build.
-->

### How Did You Get This To Happen? (Steps to Reproduce)

* Create a Release definition
* Link the Release to an existing Artifact (my artifact has spaces in it, and I can smell that this is directly related to the issue, something is not properly quoted)
* Add a Cake step to execute a .cake script inside the spaced artifact name:
  $(System.DefaultWorkingDirectory)/Artifact - Spaced/LPAMSWeb/scripts/deploy.cake
* Deploy

<!--

Can you point us to a project where this problem occurs?  i.e. a public GitHub Repo, where we can try to reproduce the problem, and help with debugging?

-->

### Output Log
<!--
When including the log information, please ensure you have run the command with --verbosity=diagnostic. It provides important information for determining an issue.

If running Cake.exe directly, the parameter is passed in as --verbosity=diagnostic
If running Cake via the bootstrapper, the parameter is -Verbosity Diagnostic

- Make sure there is no sensitive data shared.
- We need ALL output, not just what you may believe is relevant.

In order to rule out potential issues with the compilation of your Cake script, it would be very helpful if you could try running the script with some additional parameters.

The first, when compiling on Windows, would be to use the -Mono flag.  This can be passed into both Cake.exe and the Bootstrapper.  This will allow verification that it isn't a problem specific to one compiler or the other.

The second, would be to use the latest version of Roslyn.  This can be done using the -Experimental flag, which again can be passed to both Cake.exe and the bootstrapper.

If you can provide the output from the above when submitting an issue, this would be tremendously useful!
-->

2017-05-24T15:23:33.6736279Z Installing Cake...
2017-05-24T15:23:33.6906296Z ##[error]The term 'C:\tfs-agent\_work\bc7a36e29\Artifact' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
2017-05-24T15:23:33.6986304Z ##[error]Could not find Cake.exe at C:\tfs-agent\_work\bc7a36e29\Artifact - Spaced\LPAMSWeb\scripts\tools\Cake\Cake.exe
2017-05-24T15:23:33.7196325Z Finishing task: Cake 
2017-05-24T15:23:33.7206326Z ##[error]Task Cake failed. This caused the job to fail. Look at the logs for the task for more details.

<!--
TEMPLATE FOR FEATURE REQUESTS:

It's a blank slate, have fun!
-->

--- a/Task/Cake.ps1
+++ b/Task/Cake.ps1
@@ -80,27 +80,27 @@ if ((Test-Path $PackagePath)) {
   # Install tools in packages.config.
   Write-Host "Restoring packages...";
   if($ToolFeedUrl -ne ""){
-      Invoke-Expression "$NuGetPath install `"$PackagePath`" -ExcludeVersion -OutputDirectory `"$ToolPath`" -Source $ToolFeedUrl";
+      Invoke-Expression "&`"$NuGetPath`" install `"$PackagePath`" -ExcludeVersion -OutputDirectory `"$ToolPath`" -Source $ToolFeedUrl";
   }else{
-      Invoke-Expression "$NuGetPath install `"$PackagePath`" -ExcludeVersion -OutputDirectory `"$ToolPath`"";
+      Invoke-Expression "&`"$NuGetPath`" install `"$PackagePath`" -ExcludeVersion -OutputDirectory `"$ToolPath`"";
   }
 }
 if ((Test-Path $ModulePackagePath)) {
   # Install tools in Modules/packages.config
   Write-Host "Installing modules..."
   if($ToolFeedUrl -ne ""){
-      Invoke-Expression "$NuGetPath install `"$ModulePackagePath`" -ExcludeVersion -OutputDirectory `"$ModulePath`" -Source $ToolFeedUrl";
+      Invoke-Expression "&`"$NuGetPath`" install `"$ModulePackagePath`" -ExcludeVersion -OutputDirectory `"$ModulePath`" -Source $ToolFeedUrl";
   }else{
-      Invoke-Expression "$NuGetPath install `"$ModulePackagePath`" -ExcludeVersion -OutputDirectory `"$ModulePath`"";
+      Invoke-Expression "&`"$NuGetPath`" install `"$ModulePackagePath`" -ExcludeVersion -OutputDirectory `"$ModulePath`"";
   }
 }
 if (!(Test-Path $CakePath)) {
   # Install Cake if not part of packages.config.
   Write-Host "Installing Cake...";
   if($ToolFeedUrl -ne ""){
-      Invoke-Expression "$NuGetPath install Cake -ExcludeVersion -OutputDirectory `"$ToolPath`" -Source $ToolFeedUrl";
+      Invoke-Expression "&`"$NuGetPath`" install Cake -ExcludeVersion -OutputDirectory `"$ToolPath`" -Source $ToolFeedUrl";
   }else{
-      Invoke-Expression "$NuGetPath install Cake -ExcludeVersion -OutputDirectory `"$ToolPath`"";
+      Invoke-Expression "&`"$NuGetPath`" install Cake -ExcludeVersion -OutputDirectory `"$ToolPath`"";
   }
 }
 Pop-Location;

